pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -70,7 +76,6 @@ func (ps *PipelineSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	errs = errs.Also(validatePipelineContextVariables(ps.Tasks).ViaField("tasks"))
 	errs = errs.Also(validatePipelineContextVariables(ps.Finally).ViaField("finally"))
 	errs = errs.Also(validateExecutionStatusVariables(ps.Tasks, ps.Finally))
-	errs = errs.Also(ps.ValidateBetaFields(ctx))
```

Comment:
just to make it clear for the reviewers, this PR is moving this validation from `func (ps *PipelineSpec) Validate()` to `func (p *Pipeline) Validate()` such that the validation `ValidateBetaFields` is done during `pipeline` creation time compared to `pipelineRun`.